datasetinsights/stats/visualization/plots.py

```diff
@@ -128,6 +133,39 @@ def get_color(match):
     return colors
 
 
+def plot_bboxes3d(image, bboxes, projection=None, colors=None):
+    """ Plot an image with 3D bounding boxes
+
+    Currently this method should only be used for ground truth images, and
+    doesn't support predictions. If a list of colors is not provided as an
+    argument to this routine, the default color of green will be used.
+
+    Args:
+        image (PIL Image): a PIL image.
+        bboxes (list): a list of BBox3D objects
+        projection: The perspective projection of the camera of the camera which
+        captured the ground truth. Defaults to none. If no projection is
+        provided the method will use the identity to matrix for projection. This
+        will probably lead to inaccurate results
+        colors (list): a color list for boxes. Defaults to none. If
+        colors = None, it will default to coloring all boxes green.
+
+    Returns:
+        PIL image: a PIL image with bounding boxes drawn on it.
+    """
+    np_image = np.array(image)
+    img_height, img_width, _ = np_image.shape
+
+    if projection is None:
```

Comment:
I think `plot_bboxes3d` should fail if `projection` is not provided. The behavior of "plot" is undefined as there are no way to project corners in 3D space to 2D image pixels. Let's make `projection` a required argument here.